eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/PostRestartHtlcCleaner.scala

```diff
@@ -321,7 +321,7 @@ object PostRestartHtlcCleaner {
    */
   private def shouldFulfill(finalPacket: IncomingPaymentPacket.FinalPacket, paymentsDb: IncomingPaymentsDb): Option[ByteVector32] =
     paymentsDb.getIncomingPayment(finalPacket.add.paymentHash) match {
-      case Some(IncomingPayment(_, preimage, _, _, IncomingPaymentStatus.Received(_, _))) => Some(preimage)
+      case Some(p) if p.status.isInstanceOf[IncomingPaymentStatus.Received] => Some(p.paymentPreimage)
```

Comment:
You don't like pattern matching?

Reply:
Right, I can make that more elegant, will do.

Reply:
Done in https://github.com/ACINQ/eclair/pull/2416/commits/35947e4c6e190168edda688b38a7f0ecfcc3ec49